WooCommerce/Classes/POS/Controllers/PointOfSaleOrderController.swift

```diff
@@ -68,6 +69,15 @@ final class PointOfSaleOrderController: PointOfSaleOrderControllerProtocol {
         }))
     }
 
+    func sendOrderReceipt(order: Order, toEmailAddress: String) async {
+        do {
+            try await orderService.sendOrderReceipt(order: order, toEmailAddress: toEmailAddress)
+        } catch {
+            // TODO:
+            debugPrint("Error: \(error)")
```

Comment:
I left this unhandled for the moment, do we want to switch the `orderState` to `.error` here? As the Order might be completed just fine, but the receipt could fail after 🤔

Reply:
Probably not; we need to be really careful not to give the impression that the payent has failed, when actually only sending the receipt has.

Potentially this would be a good place for the view to handle the thrown errors somehow; then we can more easily display the error in-line with the receipt call that they relate to. This would probably need a specific view that can handle them, rather than the default button.

In the IPP version of this, the button turns into a spinner while it's sending, then a check mark when it's done. Equally, it could show the error and a retry button if it fails.

Reply:
That makes sense, I already have a logged issue for this case: https://github.com/woocommerce/woocommerce-ios/issues/14464. I'll see what design comes up with before making any UI changes here.